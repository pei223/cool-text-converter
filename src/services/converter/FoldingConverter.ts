import { warnLog } from "../../utils/logger";
import { ConvertibleMarkdownFormats } from "../consts/format";
import { MarkdownUnrecognizedError } from "../errors/types";
import { MarkdownConverter } from "./types";

export class FoldingConverter implements MarkdownConverter {
  recognize(v: string) {
    warnLog("Unexpected recognize call (Folding): ", v);
    throw new MarkdownUnrecognizedError(
      v,
      "Folding",
      "Unexpected recgonize call"
    );
  }
  getDetailedConverter(): MarkdownConverter | null {
    return null;
  }
  convertToMarkdown(v: string): string {
    return `<details><summary>サマリー</summary><div>\n\n${v}\n</div></details>`;
  }
  formatName(): ConvertibleMarkdownFormats {
    return "Folding";
  }
  formatLabel(): string {
    return "折りたたみ";
  }
}
