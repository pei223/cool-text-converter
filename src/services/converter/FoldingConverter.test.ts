import { describe, it, expect } from "vitest";
import { FoldingConverter } from "./FoldingConverter";

describe("FoldingConverter", () => {
  const converter = new FoldingConverter();

  describe("recognize", () => {
    it("always throw Error", () => {
      expect(() => converter.recognize("")).toThrow();
      expect(() => converter.recognize("hoge")).toThrow();
    });
  });
  // 中身いちいち正解かくのしんどいのでスナップショットテストにする
  it.each([
    ["Foldable text"],
    ["Foldable text\nFoldable text\nFoldable text\nFoldable text"],
  ])("converteToMarkdown(%s) - (%s)", (v: string) => {
    expect(converter.convertToMarkdown(v)).toMatchSnapshot();
  });
});
