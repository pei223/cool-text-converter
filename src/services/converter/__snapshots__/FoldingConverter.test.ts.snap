// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FoldingConverter > converteToMarkdown(Foldable text
Foldable text
Foldable text
Foldable text) - (%s) 1`] = `
"<details><summary>サマリー</summary><div>

Foldable text
Foldable text
Foldable text
Foldable text
</div></details>"
`;

exports[`FoldingConverter > converteToMarkdown(Foldable text) - (%s) 1`] = `
"<details><summary>サマリー</summary><div>

Foldable text
</div></details>"
`;
